src/services/campaignService.ts

```diff
@@ -36,9 +35,55 @@ export const fillCampaignProjects = async (params: {
       campaign.sortingField as unknown as SortingField;
   } else if (campaign.type === CampaignType.WithoutProjects) {
     // Dont add projects to this campaign type
-    return campaign;
+    return null;
+  }
+
+  return projectsQueryParams;
+};
+let projectCampaignCache: { [key: number]: string[] } | undefined;
+
+export const getAllProjectsRelatedToActiveCampaigns = async (): Promise<{
+  [key: number]: string[];
+}> => {
+  // It returns all project and campaigns( excluding manuallySelectedCampaign)
+  if (projectCampaignCache) {
+    return projectCampaignCache;
   }
+  projectCampaignCache = {};
+  const activeCampaigns = await findAllActiveCampaigns();
+  for (const campaign of activeCampaigns) {
+    const projectsQueryParams = createFetchCampaignProjectsQuery(campaign);
+    if (!projectsQueryParams) {
+      break;
+    }
+    const projectsQuery = filterProjectsQuery(projectsQueryParams);
+    const projects = await projectsQuery.getMany();
```

Comment:
Even with the cache, every 10 minutes it could be a big memory spike.

Reply:
I dont know how we can solve it with a simple query, if there is something in your mind definitely it would be better, because for campaigns that uses filter or sort, the included projects may change during the time so if want to know a project belongs to which campaign we should check all campaign's projects and it would decrease performance especially **projectBySlug** is one of webservices that frontend call it almost many times.

And campaigns that uses filter and sort just have 10 projects so the cached object would not be a large object so I thought maybe it's better to store it in memory instead of redis ( because it's already complicated this part, I though it  may make it more complicated